components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java

```diff
@@ -450,6 +454,25 @@ private void handleEmailVerifiedClaim(Map<String, Object> returnClaims) {
         }
     }
 
+    private void handleRolesClaim(Map<String, Object> returnClaims) {
+
+        if (returnClaims.containsKey(ROLES) && IdentityUtil.isGroupsVsRolesSeparationImprovementsEnabled()) {
+            String delim = "";
+            String separator = ",";
+            String[] roles = returnClaims.get(ROLES).toString().split(separator);
+            StringBuffer multiValuedStringBf = new StringBuffer();
+
+            for (String role : roles) {
+                if (UserCoreConstants.INTERNAL_DOMAIN.equalsIgnoreCase(IdentityUtil.extractDomainFromName(role))) {
+                    role = UserCoreUtil.removeDomainFromName(role);
+                }
+                multiValuedStringBf.append(delim).append(role);
+                delim = separator;
+            }
```

Comment:
we can use `StringUtils.join(roles, ",")` here

Reply:
we can use Java 8 filters to get the roles without the domain

Reply:
Fixed